cmd/vela-makisu/registry.go

```diff
@@ -0,0 +1,209 @@
+// Copyright (c) 2020 Target Brands, Inr. All rights reserved.
+//
+// Use of this source code is governed by the LICENSE file in this repository.
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/afero"
+	"github.com/uber/makisu/lib/registry"
+	"github.com/urfave/cli/v2"
+)
+
+const (
+	// dockerHubConf represents the config that provides authentication to docker.index.io.
+	dockerHubConf = `{
+		"index.docker.io": {
+      ".*": {
+         "security": {
+            "tls": {
+               "client": {
+                  "disabled": false
+               }
+            },
+            "basic": {
+               "username": "",
+               "password": ""
+            }
+         }
+      }
+   }		
+ }`
+
+	// mirrorConf represents the config that provides authentication to a registry mirror.
+	mirrorConf = `{
+		"%s": {
+      ".*": {
+         "security": {
+            "tls": {
+               "client": {
+                  "disabled": false
+               }
+            },
+            "basic": {
+               "username": "",
+               "password": ""
+            }
+         }
+      }
+   }		
+ }`
+
+	// registryConf represents the config that provides authentication to a registry.
+	registryConf = `{
+		"%s": {
+      ".*": {
+         "security": {
+            "tls": {
+               "client": {
+                  "disabled": false
+               }
+            },
+            "basic": {
+               "username": "%s",
+               "password": "%s"
+            }
+         }
+      }
+   }		
+ }`
+)
+
+// Registry represents the input parameters for the plugin.
+type Registry struct {
+	// full url to Docker Registry
+	Addr string
```

Comment:
I wonder if we should change this also to `Name` instead of `Addr`?

If not, we should at least update the comment since it's not accurate 👍 

We aren't actually providing a full url or address to the registry.. we're providing the name of it since it doesn't have any protocol or scheme.

i.e. `https://`, `http://` etc.

Reply:
Can update